sockeye/train.py

```diff
@@ -188,6 +188,7 @@ def main():
 
         # create data iterators
         max_seq_len_source, max_seq_len_target = args.max_seq_len
+        batch_num_devices = 1 if args.use_cpu else sum(-di if di < 0 else 1 for di in args.device_ids)
```

Comment:
I looked there first and it appeared that the GPU acquisition logic is pretty tightly coupled to identifying and locking actual devices rather than just finding a number.

Reply:
it should be possible to use `1 if args.use_cpu else len(expand_requested_device_ids(args.device_ids))`